src/main/kotlin/com/devfloor/untitled/article/application/ArticleFacade.kt

```diff
@@ -12,8 +15,33 @@ class ArticleFacade(
     private val articleService: ArticleService,
     private val articleHashtagService: ArticleHashtagService,
     private val articleFavoriteService: ArticleFavoriteService,
-    private val articleOptionService: ArticleOptionService
+    private val articleOptionService: ArticleOptionService,
+    private val optionService: OptionService,
+    private val hashtagService: HashtagService
 ) {
+    @Transactional
+    fun create(articleRequest: ArticleRequest, user: User): Long {
+        val article = articleService.create(articleRequest.toArticle(articleRequest, user))
+        if (articleRequest.options.isNotEmpty()) {
+            optionService.showAllByOptionType(articleRequest.options)
+                .let { articleOptionService.createAllByOptions(article, it) }
+        }
+        if (articleRequest.hashtags.isNotEmpty()) {
+            articleRequest.hashtags.forEach {
+                if (hashtagService.existsByName(it)) {
+                    hashtagService.create(it).let {
+                        articleHashtagService.createByHashtag(article, it)
+                    }
+                } else {
+                    hashtagService.showByName((it)).let {
+                        articleHashtagService.createByHashtag(article, it)
+                    }
+                }
+            }
+        }
```

Comment:
`articleRequest.hashtags`가 empty인 경우에는 emptyList이기 때문에 forEach loop 자체가 실행되지 않습니다. 따라서, emptyList인지를 판별하는 조건문은 불필요해 보입니다 ! 또한 중복되는 코드를 아래처럼 줄일 수 있지 않을까 합니다 !

아, 그리고 위의 31번째 줄은 `hashtagService`에 "이미 존재하는 경우"에 또 추가해주는 것으로 보여, 제가 해당 로직에 `!`를 붙여주긴 했는데요. 혹시 이 로직이 원래 의도하시던 로직이 맞나요!?
```suggestion
        articleRequest.hashtags.forEach {
            if(!hashtagService.existsByName(it))
                hashtagService.create(it)
            articleHashtagService.createByHashtag(article, Hashtag(it))
        }
```

Reply:
1. forEach가 emptyList일 때 실행되지 않군요! 조건문 제거하겠습니다.
2. 네! 31번쨰줄 ! 붙이는게 맞습니다! 허나 32번째 줄에 articleHashtag를 생성할 때 Hashtag(it)은 해시태그를 새로 생성하여 사용하는 것 아닌가요? 기존에 해시태그가 있을 경우 hashtagId를 가져와야되는데 형님처럼하면 새로운 id가 생성되는거 아닌가용?? 여쭤봅니당!

Reply:
생각해보니까 그렇네요 !! 이미 저장되어있는 hashtag와 관계를 맺도록 해야하는데 그 부분을 미처 생각하지 못했네요 !
그러면 아래와 같이 코드를 변경할 수 있을거같아요 !

hashtag 변수를 새로 선언하지 않으려면 `(if ... else ...).let { ... }` 요렇게 묶어서 쓸수도 있긴한데 저 코드가 더 깔끔해보여서 저걸 추천해드립니다 !
원래 준형님이 작성하셨던 if else 코드나 중복을 줄인 제 코드 중 가독성이 좋다고 판단하시는거로 사용하시면 될 것 같아요~

```suggestion
        articleRequest.hashtags.forEach {
            val hashtag = if (hashtagService.existsByName(it))
                hashtagService.create(it)
            else hashtagService.showByName(it)
            articleHashtagService.createByHashtag(article, hashtag)
        }
```

Reply:
형님께 더 깔끔해보입니다 수정하겠습니당! 감사합니당!